spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -107,6 +107,14 @@ default boolean useOriginalInput() {
         return false;
     }
 
+    /**
+     * Whether we should include tool's output in agent response
+     * @return true/false
+     */
+    default boolean includeOutputInAgentResponse() {
```

Comment:
This parameter is only for telling agent framework to save result in agent response. To keep the tool responsibility clear, I would suggest keep this parameter in ToolSpec, rather than Tool. A Tool just simple provide some function and output some data. Where to store the Tool response should be not coupled with Tool.

Reply:
The reason put the parameter in Tool interface is based on comments here https://github.com/opensearch-project/ml-commons/pull/1660#discussion_r1401135491

To combine two level of parameters one is toolSpec level, another is runtime level together.

Reply:
@jngz-es @ylwu-amzn how about we support toolSpec only and ignore runtime override for this parameter for now?

Reply:
thanks, remove it from tool interface.